{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "af7e1206-2f20-4eda-b7cf-7554900447de",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Registered S3 method overwritten by 'SeuratDisk':\n",
      "  method            from  \n",
      "  as.sparse.H5Group Seurat\n",
      "\n",
      "Attaching SeuratObject\n",
      "\n",
      "Loading required package: SummarizedExperiment\n",
      "\n",
      "Loading required package: MatrixGenerics\n",
      "\n",
      "Loading required package: matrixStats\n",
      "\n",
      "\n",
      "Attaching package: ‘MatrixGenerics’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:matrixStats’:\n",
      "\n",
      "    colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,\n",
      "    colCounts, colCummaxs, colCummins, colCumprods, colCumsums,\n",
      "    colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,\n",
      "    colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,\n",
      "    colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,\n",
      "    colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,\n",
      "    colWeightedMeans, colWeightedMedians, colWeightedSds,\n",
      "    colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,\n",
      "    rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,\n",
      "    rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,\n",
      "    rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,\n",
      "    rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,\n",
      "    rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,\n",
      "    rowWeightedMads, rowWeightedMeans, rowWeightedMedians,\n",
      "    rowWeightedSds, rowWeightedVars\n",
      "\n",
      "\n",
      "Loading required package: GenomicRanges\n",
      "\n",
      "Loading required package: stats4\n",
      "\n",
      "Loading required package: BiocGenerics\n",
      "\n",
      "\n",
      "Attaching package: ‘BiocGenerics’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    anyDuplicated, append, as.data.frame, basename, cbind, colnames,\n",
      "    dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,\n",
      "    grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,\n",
      "    union, unique, unsplit, which.max, which.min\n",
      "\n",
      "\n",
      "Loading required package: S4Vectors\n",
      "\n",
      "\n",
      "Attaching package: ‘S4Vectors’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Loading required package: IRanges\n",
      "\n",
      "Loading required package: GenomeInfoDb\n",
      "\n",
      "Loading required package: Biobase\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: ‘Biobase’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:MatrixGenerics’:\n",
      "\n",
      "    rowMedians\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:matrixStats’:\n",
      "\n",
      "    anyMissing, rowMedians\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: ‘SummarizedExperiment’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:SeuratObject’:\n",
      "\n",
      "    Assays\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:Seurat’:\n",
      "\n",
      "    Assays\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "library(SeuratDisk)\n",
    "library(Seurat)\n",
    "library(SingleCellExperiment)\n",
    "library(SingleR)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "939c61a0-e8b1-4ec5-9538-0efc904fba8e",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message:\n",
      "“Unknown file type: h5ad”\n",
      "Warning message:\n",
      "“'assay' not set, setting to 'RNA'”\n",
      "Creating h5Seurat file for version 3.1.5.9900\n",
      "\n",
      "Adding X as scale.data\n",
      "\n",
      "Adding raw/X as data\n",
      "\n",
      "Adding raw/X as counts\n",
      "\n",
      "Adding meta.features from raw/var\n",
      "\n",
      "Adding dispersions-0 from scaled feature-level metadata\n",
      "\n",
      "Adding dispersions-1 from scaled feature-level metadata\n",
      "\n",
      "Adding dispersions-2 from scaled feature-level metadata\n",
      "\n",
      "Adding dispersions-3 from scaled feature-level metadata\n",
      "\n",
      "Merging dispersions_norm from scaled feature-level metadata\n",
      "\n",
      "Merging highly_variable from scaled feature-level metadata\n",
      "\n",
      "Merging mean_counts from scaled feature-level metadata\n",
      "\n",
      "Merging means from scaled feature-level metadata\n",
      "\n",
      "Merging mt from scaled feature-level metadata\n",
      "\n",
      "Merging n_cells from scaled feature-level metadata\n",
      "\n",
      "Merging n_cells_by_counts from scaled feature-level metadata\n",
      "\n",
      "Merging pct_dropout_by_counts from scaled feature-level metadata\n",
      "\n",
      "Merging ribo from scaled feature-level metadata\n",
      "\n",
      "Merging total_counts from scaled feature-level metadata\n",
      "\n",
      "Validating h5Seurat file\n",
      "\n",
      "Initializing RNA with data\n",
      "\n",
      "Adding counts for RNA\n",
      "\n",
      "Adding scale.data for RNA\n",
      "\n",
      "Adding feature-level metadata for RNA\n",
      "\n",
      "Adding command information\n",
      "\n",
      "Adding cell-level metadata\n",
      "\n",
      "Adding miscellaneous information\n",
      "\n",
      "Adding tool-specific results\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "\n",
       "  CLP   CMP Flk2n Flk2p   GMP   HSC   MEP \n",
       "12250  7633  9362  5926  6431 27477  7031 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#Convert(\"/store/Projects/pantovski/Circadian/scRNA-seq/7am/write/Ambient_RNA_Cleaned/VIIam_join_inner_raw_SingleR_lognorm_200_3_10_5_scrublet_no_mt_2_UMI_5_percent_regress.h5ad\", dest = \"h5seurat\", overwrite = TRUE)\n",
    "#Convert(\"/store/Projects/pantovski/Circadian/scRNA-seq/11am/write/Ambient_RNA_Cleaned/XIam_join_inner_raw_SingleR_lognorm_200_3_10_5_scrublet_no_mt_2_UMI_5_percent_regress.h5ad\", dest = \"h5seurat\", overwrite = TRUE)\n",
    "#Convert(\"/store/Projects/pantovski/Circadian/scRNA-seq/7pm/write/Ambient_RNA_Cleaned/VIIpm_join_inner_raw_SingleR_lognorm_200_3_10_5_scrublet_no_mt_2_UMI_5_percent_regress.h5ad\", dest = \"h5seurat\", overwrite = TRUE)\n",
    "#Convert(\"/store/Projects/pantovski/Circadian/scRNA-seq/11pm/write/Ambient_RNA_Cleaned/XIpm_join_inner_raw_SingleR_lognorm_200_3_10_5_scrublet_no_mt_2_UMI_5_percent_regress.h5ad\", dest = \"h5seurat\", overwrite = TRUE)\n",
    "\n",
    "Convert(\"/store/Projects/pantovski/Circadian/scRNA-seq/Combined/write/Uncleaned/All_together_all_hvgo_MNN_hvgo_only_lognorm.h5ad\", dest = \"h5seurat\", overwrite = TRUE)\n",
    "\n",
    "All <- LoadH5Seurat(\"/store/Projects/pantovski/Circadian/scRNA-seq/Combined/write/Uncleaned/All_together_all_hvgo_MNN_hvgo_only_lognorm.h5seurat\")\n",
    "All.sce <- as.SingleCellExperiment(All)\n",
    "\n",
    "HSPC_simple_young_SingleR_ref_8ct_matrix <- read.csv(\"/store/Projects/pantovski/Pre-processing_clustering_annotation/HSPC_simple_young_SingleR_ref_8ct_02182022.csv\", row.names = \"gene_name\")\n",
    "HSPC_simple_young_SingleR_ref_8ct.sce = as.SingleCellExperiment(CreateSeuratObject(HSPC_simple_young_SingleR_ref_8ct_matrix, project = 'HSPC_simple_young'))\n",
    "\n",
    "All_pred <- SingleR(test=All.sce, ref=HSPC_simple_young_SingleR_ref_8ct.sce, labels = HSPC_simple_young_SingleR_ref_8ct.sce$orig.ident)\n",
    "table(All_pred$labels)\n",
    "write.csv(All_pred, file = \"/store/Projects/pantovski/Circadian/scRNA-seq/Combined/write/Uncleaned/All_together_all_hvgo_MNN_hvgo_only_lognorm_cell_types.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "69a63980-c536-4826-952c-d0003ca35ef4",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message:\n",
      "“Unknown file type: h5ad”\n",
      "Warning message:\n",
      "“'assay' not set, setting to 'RNA'”\n",
      "Creating h5Seurat file for version 3.1.5.9900\n",
      "\n",
      "Adding X as scale.data\n",
      "\n",
      "Adding raw/X as data\n",
      "\n",
      "Adding raw/X as counts\n",
      "\n",
      "Adding meta.features from raw/var\n",
      "\n",
      "Adding dispersions-0 from scaled feature-level metadata\n",
      "\n",
      "Adding dispersions-1 from scaled feature-level metadata\n",
      "\n",
      "Adding dispersions-2 from scaled feature-level metadata\n",
      "\n",
      "Adding dispersions-3 from scaled feature-level metadata\n",
      "\n",
      "Merging dispersions_norm from scaled feature-level metadata\n",
      "\n",
      "Merging highly_variable from scaled feature-level metadata\n",
      "\n",
      "Merging mean_counts from scaled feature-level metadata\n",
      "\n",
      "Merging means from scaled feature-level metadata\n",
      "\n",
      "Merging mt from scaled feature-level metadata\n",
      "\n",
      "Merging n_cells from scaled feature-level metadata\n",
      "\n",
      "Merging n_cells_by_counts from scaled feature-level metadata\n",
      "\n",
      "Merging pct_dropout_by_counts from scaled feature-level metadata\n",
      "\n",
      "Merging ribo from scaled feature-level metadata\n",
      "\n",
      "Merging total_counts from scaled feature-level metadata\n",
      "\n",
      "Validating h5Seurat file\n",
      "\n",
      "Initializing RNA with data\n",
      "\n",
      "Adding counts for RNA\n",
      "\n",
      "Adding scale.data for RNA\n",
      "\n",
      "Adding feature-level metadata for RNA\n",
      "\n",
      "Adding command information\n",
      "\n",
      "Adding cell-level metadata\n",
      "\n",
      "Adding miscellaneous information\n",
      "\n",
      "Adding tool-specific results\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "\n",
       "  CLP   CMP Flk2n Flk2p   GMP   HSC   MEP \n",
       " 8408  5122  6212   709  7000 35170 13489 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "Convert(\"/store/Projects/pantovski/Circadian/scRNA-seq/Combined/write/Uncleaned/All_together_MNN_2_UMI_5_percent_lognorm.h5ad\", dest = \"h5seurat\", overwrite = TRUE)\n",
    "\n",
    "All <- LoadH5Seurat(\"/store/Projects/pantovski/Circadian/scRNA-seq/Combined/write/Uncleaned/All_together_MNN_2_UMI_5_percent_lognorm.h5seurat\")\n",
    "All.sce <- as.SingleCellExperiment(All)\n",
    "\n",
    "HSPC_simple_young_SingleR_ref_8ct_matrix <- read.csv(\"/store/Projects/pantovski/Pre-processing_clustering_annotation/HSPC_simple_young_SingleR_ref_8ct_02182022.csv\", row.names = \"gene_name\")\n",
    "HSPC_simple_young_SingleR_ref_8ct.sce = as.SingleCellExperiment(CreateSeuratObject(HSPC_simple_young_SingleR_ref_8ct_matrix, project = 'HSPC_simple_young'))\n",
    "\n",
    "All_pred <- SingleR(test=All.sce, ref=HSPC_simple_young_SingleR_ref_8ct.sce, labels = HSPC_simple_young_SingleR_ref_8ct.sce$orig.ident)\n",
    "table(All_pred$labels)\n",
    "write.csv(All_pred, file = \"/store/Projects/pantovski/Circadian/scRNA-seq/Combined/write/Uncleaned/All_together_MNN_2_UMI_5_percent_lognorm_cell_types.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "48ba9deb-8d07-4c5b-8e0c-a2f0b3e6d451",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message:\n",
      "“Unknown file type: h5ad”\n",
      "Warning message:\n",
      "“'assay' not set, setting to 'RNA'”\n",
      "Creating h5Seurat file for version 3.1.5.9900\n",
      "\n",
      "Adding X as scale.data\n",
      "\n",
      "Adding raw/X as data\n",
      "\n",
      "Adding raw/X as counts\n",
      "\n",
      "Adding meta.features from raw/var\n",
      "\n",
      "Adding dispersions-0 from scaled feature-level metadata\n",
      "\n",
      "Adding dispersions-1 from scaled feature-level metadata\n",
      "\n",
      "Adding dispersions-2 from scaled feature-level metadata\n",
      "\n",
      "Adding dispersions-3 from scaled feature-level metadata\n",
      "\n",
      "Merging dispersions_norm from scaled feature-level metadata\n",
      "\n",
      "Merging highly_variable from scaled feature-level metadata\n",
      "\n",
      "Merging mean_counts from scaled feature-level metadata\n",
      "\n",
      "Merging means from scaled feature-level metadata\n",
      "\n",
      "Merging mt from scaled feature-level metadata\n",
      "\n",
      "Merging n_cells from scaled feature-level metadata\n",
      "\n",
      "Merging n_cells_by_counts from scaled feature-level metadata\n",
      "\n",
      "Merging pct_dropout_by_counts from scaled feature-level metadata\n",
      "\n",
      "Merging ribo from scaled feature-level metadata\n",
      "\n",
      "Merging total_counts from scaled feature-level metadata\n",
      "\n",
      "Validating h5Seurat file\n",
      "\n",
      "Initializing RNA with data\n",
      "\n",
      "Adding counts for RNA\n",
      "\n",
      "Adding scale.data for RNA\n",
      "\n",
      "Adding feature-level metadata for RNA\n",
      "\n",
      "Adding command information\n",
      "\n",
      "Adding cell-level metadata\n",
      "\n",
      "Adding miscellaneous information\n",
      "\n",
      "Adding tool-specific results\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "\n",
       "  CLP   CMP Flk2n Flk2p   GMP   HSC   MEP \n",
       "11030  6244  8188  1547  6433 31255 11413 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "Convert(\"/store/Projects/pantovski/Circadian/scRNA-seq/Combined/write/Uncleaned/All_together_MNN_2_UMI_2_percent_lognorm.h5ad\", dest = \"h5seurat\", overwrite = TRUE)\n",
    "\n",
    "All <- LoadH5Seurat(\"/store/Projects/pantovski/Circadian/scRNA-seq/Combined/write/Uncleaned/All_together_MNN_2_UMI_2_percent_lognorm.h5seurat\")\n",
    "All.sce <- as.SingleCellExperiment(All)\n",
    "\n",
    "HSPC_simple_young_SingleR_ref_8ct_matrix <- read.csv(\"/store/Projects/pantovski/Pre-processing_clustering_annotation/HSPC_simple_young_SingleR_ref_8ct_02182022.csv\", row.names = \"gene_name\")\n",
    "HSPC_simple_young_SingleR_ref_8ct.sce = as.SingleCellExperiment(CreateSeuratObject(HSPC_simple_young_SingleR_ref_8ct_matrix, project = 'HSPC_simple_young'))\n",
    "\n",
    "All_pred <- SingleR(test=All.sce, ref=HSPC_simple_young_SingleR_ref_8ct.sce, labels = HSPC_simple_young_SingleR_ref_8ct.sce$orig.ident)\n",
    "table(All_pred$labels)\n",
    "write.csv(All_pred, file = \"/store/Projects/pantovski/Circadian/scRNA-seq/Combined/write/Uncleaned/All_together_MNN_2_UMI_2_percent_lognorm_cell_types.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "a4cc7c7c-ccec-48c1-97be-b99778f4cf85",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message:\n",
      "“Unknown file type: h5ad”\n",
      "Warning message:\n",
      "“'assay' not set, setting to 'RNA'”\n",
      "Creating h5Seurat file for version 3.1.5.9900\n",
      "\n",
      "Adding X as scale.data\n",
      "\n",
      "Adding raw/X as data\n",
      "\n",
      "Adding raw/X as counts\n",
      "\n",
      "Adding meta.features from raw/var\n",
      "\n",
      "Adding dispersions-0 from scaled feature-level metadata\n",
      "\n",
      "Adding dispersions-1 from scaled feature-level metadata\n",
      "\n",
      "Adding dispersions-2 from scaled feature-level metadata\n",
      "\n",
      "Adding dispersions-3 from scaled feature-level metadata\n",
      "\n",
      "Merging dispersions_norm from scaled feature-level metadata\n",
      "\n",
      "Merging highly_variable from scaled feature-level metadata\n",
      "\n",
      "Merging mean_counts from scaled feature-level metadata\n",
      "\n",
      "Merging means from scaled feature-level metadata\n",
      "\n",
      "Merging mt from scaled feature-level metadata\n",
      "\n",
      "Merging n_cells from scaled feature-level metadata\n",
      "\n",
      "Merging n_cells_by_counts from scaled feature-level metadata\n",
      "\n",
      "Merging pct_dropout_by_counts from scaled feature-level metadata\n",
      "\n",
      "Merging ribo from scaled feature-level metadata\n",
      "\n",
      "Merging total_counts from scaled feature-level metadata\n",
      "\n",
      "Validating h5Seurat file\n",
      "\n",
      "Initializing RNA with data\n",
      "\n",
      "Adding counts for RNA\n",
      "\n",
      "Adding scale.data for RNA\n",
      "\n",
      "Adding feature-level metadata for RNA\n",
      "\n",
      "Adding command information\n",
      "\n",
      "Adding cell-level metadata\n",
      "\n",
      "Adding miscellaneous information\n",
      "\n",
      "Adding tool-specific results\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "\n",
       "  CLP   CMP Flk2n Flk2p   GMP   HSC   MEP \n",
       "12664  8877  8398  2526  6675 27454  9516 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "Convert(\"/store/Projects/pantovski/Circadian/scRNA-seq/Combined/write/Uncleaned/All_together_MNN_1_UMI_2_percent_lognorm.h5ad\", dest = \"h5seurat\", overwrite = TRUE)\n",
    "\n",
    "All <- LoadH5Seurat(\"/store/Projects/pantovski/Circadian/scRNA-seq/Combined/write/Uncleaned/All_together_MNN_1_UMI_2_percent_lognorm.h5seurat\")\n",
    "All.sce <- as.SingleCellExperiment(All)\n",
    "\n",
    "HSPC_simple_young_SingleR_ref_8ct_matrix <- read.csv(\"/store/Projects/pantovski/Pre-processing_clustering_annotation/HSPC_simple_young_SingleR_ref_8ct_02182022.csv\", row.names = \"gene_name\")\n",
    "HSPC_simple_young_SingleR_ref_8ct.sce = as.SingleCellExperiment(CreateSeuratObject(HSPC_simple_young_SingleR_ref_8ct_matrix, project = 'HSPC_simple_young'))\n",
    "\n",
    "All_pred <- SingleR(test=All.sce, ref=HSPC_simple_young_SingleR_ref_8ct.sce, labels = HSPC_simple_young_SingleR_ref_8ct.sce$orig.ident)\n",
    "table(All_pred$labels)\n",
    "write.csv(All_pred, file = \"/store/Projects/pantovski/Circadian/scRNA-seq/Combined/write/Uncleaned/All_together_MNN_1_UMI_2_percent_lognorm_cell_types.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "e077e0df-a467-46f7-9802-77ef2f26e984",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message:\n",
      "“Unknown file type: h5ad”\n",
      "Warning message:\n",
      "“'assay' not set, setting to 'RNA'”\n",
      "Creating h5Seurat file for version 3.1.5.9900\n",
      "\n",
      "Adding X as scale.data\n",
      "\n",
      "Adding raw/X as data\n",
      "\n",
      "Adding raw/X as counts\n",
      "\n",
      "Adding meta.features from raw/var\n",
      "\n",
      "Adding dispersions-0 from scaled feature-level metadata\n",
      "\n",
      "Adding dispersions-1 from scaled feature-level metadata\n",
      "\n",
      "Adding dispersions-2 from scaled feature-level metadata\n",
      "\n",
      "Adding dispersions-3 from scaled feature-level metadata\n",
      "\n",
      "Adding dispersions_norm from scaled feature-level metadata\n",
      "\n",
      "Adding highly_variable from scaled feature-level metadata\n",
      "\n",
      "Adding mean_counts from scaled feature-level metadata\n",
      "\n",
      "Adding means from scaled feature-level metadata\n",
      "\n",
      "Adding mt from scaled feature-level metadata\n",
      "\n",
      "Adding n_cells from scaled feature-level metadata\n",
      "\n",
      "Adding n_cells_by_counts from scaled feature-level metadata\n",
      "\n",
      "Adding pct_dropout_by_counts from scaled feature-level metadata\n",
      "\n",
      "Adding ribo from scaled feature-level metadata\n",
      "\n",
      "Adding total_counts from scaled feature-level metadata\n",
      "\n",
      "Validating h5Seurat file\n",
      "\n",
      "Initializing RNA with data\n",
      "\n",
      "Adding counts for RNA\n",
      "\n",
      "Adding scale.data for RNA\n",
      "\n",
      "Adding feature-level metadata for RNA\n",
      "\n",
      "Adding command information\n",
      "\n",
      "Adding cell-level metadata\n",
      "\n",
      "Adding miscellaneous information\n",
      "\n",
      "Adding tool-specific results\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "\n",
       "  CLP   CMP Flk2n Flk2p   GMP   HSC   MEP \n",
       "13328  9512 12184  7777  5849 19295  8165 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "Convert(\"/store/Projects/pantovski/Circadian/scRNA-seq/Combined/write/Uncleaned/All_together_all_hvgo_MNN_lognorm.h5ad\", dest = \"h5seurat\", overwrite = TRUE)\n",
    "\n",
    "All <- LoadH5Seurat(\"/store/Projects/pantovski/Circadian/scRNA-seq/Combined/write/Uncleaned/All_together_all_hvgo_MNN_lognorm.h5seurat\")\n",
    "All.sce <- as.SingleCellExperiment(All)\n",
    "\n",
    "HSPC_simple_young_SingleR_ref_8ct_matrix <- read.csv(\"/store/Projects/pantovski/Pre-processing_clustering_annotation/HSPC_simple_young_SingleR_ref_8ct_02182022.csv\", row.names = \"gene_name\")\n",
    "HSPC_simple_young_SingleR_ref_8ct.sce = as.SingleCellExperiment(CreateSeuratObject(HSPC_simple_young_SingleR_ref_8ct_matrix, project = 'HSPC_simple_young'))\n",
    "\n",
    "All_pred <- SingleR(test=All.sce, ref=HSPC_simple_young_SingleR_ref_8ct.sce, labels = HSPC_simple_young_SingleR_ref_8ct.sce$orig.ident)\n",
    "table(All_pred$labels)\n",
    "write.csv(All_pred, file = \"/store/Projects/pantovski/Circadian/scRNA-seq/Combined/write/Uncleaned/All_together_all_hvgo_MNN_cell_types.csv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "30629b3c",
   "metadata": {},
   "source": [
    "## After looking at the ratio of the different cell types produced with the different files, the <b>All_together_all_hvgo_MNN_lognorm.h5ad</b> and the <b>All_together_all_hvgo_MNN_cell_types.csv</b> were found to produce the best ratio corresponding to the ratio of cells submitted for RNA sequencing."
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3.10.8 64-bit (microsoft store)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "python",
   "pygments_lexer": "r",
   "version": "3.10.8"
  },
  "vscode": {
   "interpreter": {
    "hash": "35c8fdf2383800430be614face404230e2a04e77624b0b0b2475541fdc978597"
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
